src/plugins/kibana/public/dashboard/index.js

```diff
@@ -31,15 +31,44 @@ define(function (require) {
 
   require('ui/routes')
   .when('/dashboard', {
-    template: require('plugins/kibana/dashboard/index.html'),
     resolve: {
-      dash: function (savedDashboards, config) {
+      dash: function (savedDashboards, config, kbnUrl, Notifier) {
+        var defaultDashboard = config.get('dashboard:defaultDashboard', '');
+
+        var notify = new Notifier({
+          location: 'Dashboard'
+        });
+
+        if (defaultDashboard !== '') {
+          return savedDashboards.get(defaultDashboard)
+            .then(function (result) {
+
+              // If the default dashboard has been found redirect to that dashboard
+              var dashboardUrl = savedDashboards.urlFor(result.id).substring(1);
+              kbnUrl.change(dashboardUrl);
+            })
+            .catch(function (error) {
+
+              // If the given config dashboard has not been found redirect to making new dashboard
+              notify.error(error);
+              kbnUrl.change('/dashboard/new');
+            });
+        }
+
+        kbnUrl.change('/dashboard/new');
+      }
+    }
+  })
+  .when('/dashboard/new', {
```

Comment:
I would prefer that we didn't have a route for `/dashboard/new`. I think we should simply be redirecting requests from `/dashboard` to `/dashboard/:id` if a default dashboard is defined.

I think we should instead use a query string to tell the route handler for `/dashboard` that we are not interested in redirecting but definitely want a blank dashboard, something like `?create=true` or `?default=false`.